### DISCUSSION TITLE
16.0.1 minimum required openssl version

### DISCUSSION THREAD

**Discussion by GK-774** [jcorporation/myMPD#1322] [Q&A] [answered]
On an EL9 system (openssl 3.0.7) mympd 16.0.1 compiles fine, on EL8 (openssl 1.1.1k) I get an error (the previous 15.0.2 compiles without errors). Which minimum openssl version is now required?

The error is:
In file included from /usr/include/openssl/x509.h:25,
                 from /usr/include/openssl/pem.h:17,
                 from /home/gk/rpmbuild/BUILD/mympd-16.0.1/src/lib/cert.c:26:
/home/gk/rpmbuild/BUILD/mympd-16.0.1/src/lib/cert.c: In function 'generate_keypair':
/home/gk/rpmbuild/BUILD/mympd-16.0.1/src/lib/cert.c:666:51: error: conversion to 'int' from 'unsigned int' may change the sign of the result [-Werror=sign-conversion]
         if (EVP_PKEY_CTX_set_rsa_keygen_bits(ctx, key_bits) <= 0) {
                                                   ^~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/mympd.dir/build.make:174: src/CMakeFiles/mympd.dir/lib/cert.c.o] Error 1

**Comment by jcorporation** [comment]
This is an implicit conversion bug. Will be fixed with next release (16.0.2).

**Reply by jcorporation** [reply]
Should be fixed in devel. Can you test? I have no access to a linux system with such an old OpenSSL library.

**Comment by GK-774** [comment, marked as answer]
Yes, I can confirm it´s fixed. Many thanks!

The last 1.1.1 openssl is from May 2023, enterprise distributions tend to have a long runtime and there are just so many software packages that depend on a specific openssl version that it usually never gets changed (except for bug and security fixes).
